src/datatypes2/src/data_type.rs

```diff
@@ -61,66 +61,67 @@ pub enum ConcreteDataType {
     List(ListType),
 }
 
-// TODO(yingwen): Consider moving these methods to the DataType trait.
+// TODO(yingwen): Refactor these `is_xxx()` methods, such as adding a `properties()` method
+// returning all these properties to the `DataType` trait
 impl ConcreteDataType {
-    // pub fn is_float(&self) -> bool {
-    //     matches!(
-    //         self,
-    //         ConcreteDataType::Float64(_) | ConcreteDataType::Float32(_)
-    //     )
-    // }
-
-    // pub fn is_boolean(&self) -> bool {
-    //     matches!(self, ConcreteDataType::Boolean(_))
-    // }
-
-    // pub fn stringifiable(&self) -> bool {
-    //     matches!(
-    //         self,
-    //         ConcreteDataType::String(_)
-    //             | ConcreteDataType::Date(_)
-    //             | ConcreteDataType::DateTime(_)
-    //             | ConcreteDataType::Timestamp(_)
-    //     )
-    // }
-
-    // pub fn is_signed(&self) -> bool {
-    //     matches!(
-    //         self,
-    //         ConcreteDataType::Int8(_)
-    //             | ConcreteDataType::Int16(_)
-    //             | ConcreteDataType::Int32(_)
-    //             | ConcreteDataType::Int64(_)
-    //             | ConcreteDataType::Date(_)
-    //             | ConcreteDataType::DateTime(_)
-    //             | ConcreteDataType::Timestamp(_)
-    //     )
-    // }
-
-    // pub fn is_unsigned(&self) -> bool {
-    //     matches!(
-    //         self,
-    //         ConcreteDataType::UInt8(_)
-    //             | ConcreteDataType::UInt16(_)
-    //             | ConcreteDataType::UInt32(_)
-    //             | ConcreteDataType::UInt64(_)
-    //     )
-    // }
-
-    // pub fn numerics() -> Vec<ConcreteDataType> {
-    //     vec![
-    //         ConcreteDataType::int8_datatype(),
-    //         ConcreteDataType::int16_datatype(),
-    //         ConcreteDataType::int32_datatype(),
-    //         ConcreteDataType::int64_datatype(),
-    //         ConcreteDataType::uint8_datatype(),
-    //         ConcreteDataType::uint16_datatype(),
-    //         ConcreteDataType::uint32_datatype(),
-    //         ConcreteDataType::uint64_datatype(),
-    //         ConcreteDataType::float32_datatype(),
-    //         ConcreteDataType::float64_datatype(),
-    //     ]
-    // }
+    pub fn is_float(&self) -> bool {
+        matches!(
+            self,
+            ConcreteDataType::Float64(_) | ConcreteDataType::Float32(_)
+        )
+    }
+
+    pub fn is_boolean(&self) -> bool {
+        matches!(self, ConcreteDataType::Boolean(_))
+    }
+
+    pub fn stringifiable(&self) -> bool {
```

Comment:
```suggestion
    pub fn is_stringifiable(&self) -> bool {
```

To keep consistent with other predicates.